GoogleChromeLabs/ps-analysis-tool
# Fix: Empty cookie list

## Description

Cookie list and the popup summary displays no cookies intermittently. Changes in this PR fixes this issue by moving `tabUrl` management to state providers to display layer.

## Relevant Technical Choices
- Separate state providers for devtools and popup ui.
- Devtools state provider uses `chrome.devetools.inspectedWindow.eval` to get the tabs url reliably.
- Popup UI state provider uses previously used `getCurrentTab` utility to get a tabs url reliably.
- Popup UI state provider now calculates the cookie stats rather than the presentational component.
- Remove tab url management related functions from `CookieStore`.
- Remove event handling required for `tabUrl` management. ( State provider for devtools and pop up do that now )



## Testing Instructions

- Install the tool as mentioned in the README.md file.
- Pin the popup icon in the browser.
- Open multiple tabs with different sites and open the devtools in some of them. 
- Cookie count on the icon badge, piechart in the popup and list in the devtools should have consistent data on every tab.



## Additional Information:



## Screenshot/Screencast



#### Video of broken functionality
https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/53055971/5cae3db3-a9cb-41e3-ae51-af13b9e5ddab


---



Fixes #
